src/contracts/core/Slasher.sol

```diff
@@ -5,233 +5,475 @@ import "./SlasherStorage.sol";
+     * slashable stake to an operator set. Slashable stake for an operator set is
+     * (slashableMagnitude / sum of all slashableMagnitudes for the strategy/operator + nonSlashableMagnitude) of
+     * an operator's delegated stake.
+     */
+    struct OperatorSetMagnitudeParam {
+        OperatorSet operatorSet;
+        uint64 magnitudeDiff; 
+    }
+
+    /**
+     * @notice Input param struct used for functions queueAllocation and queueDeallocation.
+     * A structure defining a set of operator-based slashing configurations
+     * to manage slashable stake.
+     * @param strategy each magnitude param is defined within a single strategy
+     * @param operatorSetMagnitudeParams the input magnitude parameters defining the amount
+     * the operator set is able to slash an operator. This is passed as an array of
+     * operator sets and magnitudes for a given
+     */
+    struct MagnitudeAdjustmentParam {
+        IStrategy strategy;
+        OperatorSetMagnitudeParam[] operatorSetMagnitudeParams;
+    }
+
+    /**
+     * @notice Used for historical magnitude updates in mapping
+     * operator => IStrategy => avs => operatorSetId => MagnitudeUpdate[]
+     * New updates are pushed whenever queueDeallocations
+     * or completeAllocations is called.
+     */
+    struct MagnitudeUpdate {
+        uint32 timestamp;
+        uint64 magnitude;
+    }
+    
+    /**
+     * @notice Used for historical magnitude updates in mapping
+     * operator => IStrategy => TotalAndNonslashableUpdate[]
+     * New total magnitude updates are pushed whenever queueDeallocations
+     * or completeAllocations is called.
+     */
+    struct TotalAndNonslashableUpdate {
+        uint32 timestamp;
+        uint64 totalMagnitude;
+        uint64 nonSlashableMagnitude;
+    }
+
+    /// EVENTS
+    
+    event QueuedAllocation(
+        address operator,
+        IStrategy strategy,
+        OperatorSet operatorSet,
+        uint32 effectTimestamp,
+        uint64 slashableMagnitude
+    );
+    
+    event QueuedDeallocation(
+        address operator,
+        IStrategy strategy,
+        OperatorSet operatorSet,
+        uint32 effectTimestamp,
+        uint64 slashableMagnitude
+    );
+
+    event TotalAndNonSlashableMagnitudeUpdated(
+        address operator,
+        IStrategy strategy,
+        uint32 effectTimestamp,
+        uint64 nonSlashableMagnitude,
+        uint64 totalSlashableMagnitude
+    );
+
+    /// operator => strategy => avs => operatorSetId => MagnitudeUpdate[]
+    mapping(address => mapping(IStrategy => mapping(address => mapping(uint32 => MagnitudeUpdate[])))) private _magnitudeUpdates;
+
+    /// operator => strategy => TotalAndNonslashableUpdate[]
+    mapping(address => mapping(IStrategy => TotalAndNonslashableUpdate[])) private _totalMagnitudeUpdates;
+    /// operator => allocator
+    /// could perhaps use 
+    mapping(address => address) public allocatorFor;
```

Comment:
edit delegation manager here according to spec

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/623/commits/f7c9a75affd75db460bf53ff9dc6d5b8ff9d32db
Added allocator role in `OperatorDetails` struct